### PR TITLE
Add featured case study slug to open source page query 

### PR DESCRIPTION
**Pull request by jsms90**
## Fix featured case study link bug on open source page

- add `slug` to open source page query, for use in the `CaseStudyPreview` component:
image
- added prop types to the `CaseStudyPreview` component to try and help engineers catch this kind of mistake in future
- added [`airbnb-prop-types` package](https://github.com/airbnb/prop-types) for the use of their `or` validator, which doesn't come with [`prop-types`](https://reactjs.org/docs/typechecking-with-proptypes.html) by default

## Review template

- [ ] checked that this PR resolves the spec provided from trello / this description
- [ ] ensured that this PR does not introduce any obvious bugs

### CODE REVIEW

File: src/components/Common/CaseStudyCards/CaseStudyPreview.js [hunk 1]
```diff
@@ -59,4 +61,17 @@ const CaseStudyPreview = ({ isTop, caseStudy }) => {
...
 
+CaseStudyPreview.propTypes = {
+  isTop: PropTypes.bool,
+  caseStudy: PropTypes.shape({
+    title: PropTypes.string.isRequired,
+    slug: PropTypes.string.isRequired,
+    posterImage: PropTypes.object,
+    introSentence: or([
+      PropTypes.string.isRequired,
+      PropTypes.objectOf(PropTypes.string.isRequired).isRequired
```
Comment (omonk) [inline src/components/Common/CaseStudyCards/CaseStudyPreview.js]: should this not be `shape`? this just means an object with values of a string.

`PropTypes.shape({introSentence: PropTypes.string.isRequired})`

File: src/pages/open-source.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 import { StaticQuery, graphql } from 'gatsby'
 
+import getIntroSentence from '../utils/getIntroSentence'
```
Comment (omonk) [inline src/pages/open-source.js]: remove

Comment (jsms90) [reply src/pages/open-source.js]: in favour of what?

**Comment by omonk** [comment]
Also I thought package-lock.json had been removed?

**Comment by jsms90** [comment]
`package-lock.json` seems to still be in master? :woman_shrugging:

**Comment by omonk** [comment]
I seemed to have introduced it in this PR https://github.com/yldio/yld.io/commit/7ab39cb32b8b7b547144d31370796f3dfba0b51d

I will remove in the final  open-source-page PR

**Pull request merged**